sources/mongodb/helpers.py

```diff
@@ -235,9 +237,53 @@ def convert_mongo_objs(value: Any) -> Any:
         return str(value)
     if isinstance(value, _datetime.datetime):
         return ensure_pendulum_datetime(value)
+    if isinstance(value, Regex):
+        return value.try_compile().pattern
+    if isinstance(value, Timestamp):
+        date = value.as_datetime()
+        date = date.replace(tzinfo=timezone("UTC"))
+        return date
+
     return value
 
 
+def convert_arrow_columns(table: Any) -> Any:
+    """Convert the given table columns to Python types.
+
+    Args:
+        table (pyarrow.lib.Table): The table to convert.
+
+    Returns:
+        pyarrow.lib.Table: The table with the columns converted.
+    """
+    from pymongoarrow.types import _is_binary, _is_code, _is_decimal128, _is_objectid  # type: ignore
+    from dlt.common.libs.pyarrow import pyarrow
+
+    for i, field in enumerate(table.schema):
+        if _is_objectid(field.type) or _is_decimal128(field.type):
+            col_values = [str(value) for value in table[field.name]]
+            table = table.set_column(
+                i,
+                pyarrow.field(field.name, pyarrow.string()),
+                pyarrow.array(col_values, type=pyarrow.string()),
+            )
+        else:
+            type_ = None
+            if _is_binary(field.type):
+                type_ = pyarrow.binary()
```

Comment:
a regular cast is not working? I was able to cast object_id directly to binary so that should also work without using any list ie.
```
def convert_time_to_us(table):
    """map transform converting time column to microseconds (ie. from nanoseconds)"""
    import pyarrow as pa
    from pyarrow import compute as pc

    time_ns_column = table['time_col']
    time_us_column = pc.cast(time_ns_column, pa.time64('us'), safe=False)
    new_table = table.set_column(
        table.column_names.index('time_col'),
        'time_col',
        time_us_column,
    )
    return new_table
```
try to use `pc.cast(` and IMO it will work

Reply:
No, it's not working because of the same reasons I used column replacement in the #486 

If we just cast a column, it may turn all the values into `null`, or, in case of binary, it'll not load into the destination (it'll be converted, but fail on load)